### ISSUE TITLE
Mustache finds items in the definition file, but they still get logged (and marked) as wrong

### ISSUE THREAD

**Issue by larribas** [webyom/gulp-html-i18n#29]
Versions:
gulp-html-i18n@^0.8.1
mustache@^2.3.0

Gulp task:

```js
gulp.task('localize', function(){
    return gulp.src('./index.html')
        .pipe(i18n({
            langDir: './lang',
            renderEngine: 'mustache',
            failOnMissing: true,
            createLangDirs: true,
            defaultLang: 'es'
        }))
        .pipe(gulp.dest('./dist'));
});
```

Having the same keys as with the regex engine, it fails. If I set `failOnMissing` to `false` it logs all the keys as not being present.

Log output:

```
[11:10:41] index.footer.email not found in definition file! 
[11:10:41] index.footer.facebook not found in definition file! 
[11:10:41] index.footer.email not found in definition file!
[11:10:41] index.footer.facebook not found in definition file!
```

However, the page is rendered just fine.

**Comment by webyom** [comment]
@larribas could you help debug or upload your code for my reference.

**Comment by constantm** [comment]
I experience the same issue. I'm using the library along with Foundation/Panini:

```
function pages() {
  return gulp.src('src/pages/**/*.{html,hbs,handlebars}')
    .pipe(panini({
      root: 'src/pages/',
      layouts: 'src/layouts/',
      partials: 'src/partials/',
      data: 'src/data/',
      helpers: 'src/helpers/'
    }))
    .pipe(i18n({
      langDir: 'src/lang/',
      failOnMissing: false,
      delimiters: ['{${','}$}'],
      trace: true
    }))
    .pipe(gulp.dest(PATHS.dist));
}
```

and get the following error:

```
[12:07:25] Starting 'copy'...
[12:07:27] index.heading not found in definition file!
[12:07:27] Finished 'pages' after 2.02 s
```

However, it renders the file perfectly. 

I'm not using Mustache, and I'm also using `"gulp-html-i18n": "^0.8.1"`

**Comment by webyom** [comment]
@constantm could you upload your project for my debugging?

**Comment by FranciscoKnebel** [comment]
Can confirm that it is an issue with something from 0.8.1.
Setting my package.json package dependency to version 0.8.0 and reinstalling the package no longer showed an error message.

From what I saw, 0.8.1 was just some empty value checks, so it may be considering some values to be empty when in fact they are not. If I find some time later, i'll clone the repo to debug it.

**Comment by webyom** [comment]
@cherepanov  as the investigation from @FranciscoKnebel, it seems that the empty value checking of mustache brought in this issue. Could you help verify it.

**Comment by webyom** [comment]
Published `0.8.2` to fix this issue. Thx for @FranciscoKnebel

**Comment by larribas** [comment]
Thank you @webyom & @FranciscoKnebel. I'm sorry I couldn't answer before, I was out for a few days ;)
